taker-frontend/src/useEventSource.ts

```diff
@@ -1,9 +1,7 @@
-import { useToast } from "@chakra-ui/react";
-import { useEffect, useRef, useState } from "react";
+import { useEffect, useState } from "react";
 
-export function useEventSource(url: string, withCredentials?: boolean) {
+export function useEventSource(url: string, withCredentials?: boolean, setIsConnected) {
     const [source, setSource] = useState<EventSource | null>(null);
-    const [isConnected, setIsConnected] = useState<boolean>(true);
```

Comment:
Why not keep the state inside and return `isConnected` together with the event source?

Reply:
In the react docs they say move state up and pass the the set functions to the children. I don't know tbh.

Reply:
Generally yes. Move it up as far as where you need to modify it. You don't actually call `setIsConnected` outside of this hook so there isn't really any point tracking the state outside.